Backpack/BarChart/Classes/BarChartCollectionView.swift

```diff
@@ -0,0 +1,149 @@
+    }()
+    
+    fileprivate func updateSelectedMarkerPosition() {
+        if selectedMarkerBottomConstraint != nil {
+            selectedMarkerBottomConstraint?.isActive = false
+            selectedMarkerBottomConstraint = nil
+        }
+        
+        if selectedIndexPath == nil {
+            selectedMarker.isHidden = true
+        } else {
+            selectedMarker.isHidden = false
+
+            guard let selectedCell: BPKBarChartCollectionViewCell =
+                cellForItem(at: selectedIndexPath!) as? BPKBarChartCollectionViewCell
+            else {
+                return
+            }
+            let selectedBarTopPosition: CGFloat = selectedCell.barChartBar.barTopPosition
+            selectedMarkerBottomConstraint =
+                selectedMarker.bottomAnchor.constraint(equalTo: topAnchor, constant: selectedBarTopPosition)
+            selectedMarkerBottomConstraint?.isActive = true
+        }
+    }
+}
+
+extension BPKBarChartCollectionView: UICollectionViewDataSource {
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return barChartDataSource?.numberOfBarsInChart(barChartCollectionView: self) ?? 0
+    }
+    
+    public func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    public func collectionView(
+        _ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        guard let cell = dequeueReusableCell(
+            withReuseIdentifier: BPKBarChartCollectionView.cellIdentifier,
+            for: indexPath) as? BPKBarChartCollectionViewCell
+            else {
+                fatalError("No cell registered for reuse with identifier \(BPKBarChartCollectionView.cellIdentifier)")
+        }
+        cell.barChartBar.title = barChartDataSource?.titleForBar(barChartCollectionView: self,
+                                                                 atIndex: indexPath.item)
+        cell.barChartBar.subtitle = barChartDataSource?.subtitleForBar(barChartCollectionView: self,
+                                                                       atIndex: indexPath.item)
+        cell.barChartBar.fillValue = barChartDataSource?.fillValueForBar(barChartCollectionView: self,
+                                                                         atIndex: indexPath.item)
+        cell.barChartBar.valueDescription = barChartDataSource?.valueDescriptionForBar(barChartCollectionView: self,
+                                                                                       atIndex: indexPath.item)
+        cell.isSelected = selectedIndexPath == indexPath
+        return cell
+    }
+}
+
+extension BPKBarChartCollectionView: UICollectionViewDelegate {
+    public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        selectedIndexPath = indexPath
+        
+        if barChartDelegate != nil {
```

Comment:
Afaik you dont need to check this, just doing 

`barChartDelegate?.didSelect(barChart: self, index: indexPath.item)` will do nothing if `barChartDelegate` is `nil`